src/t_hash.c

```diff
@@ -128,12 +126,26 @@ mstrKind mstrFieldKind = {
 };
 static_assert(sizeof(struct ExpireMeta ) % 2 == 0, "must be even!");
 
-typedef enum CmdTimeExpRes {
-    HFE_NO_FIELD =         -2,   /* No such hash-field */
-    HFE_NO_CONDITION_MET =  0,   /* Specified NX | XX | GT | LT condition not met */
-    HFE_SET_OK =            1,   /* Expiration time set/updated as expected */
-    HFE_DELETED =           2    /* Field deleted because the specified time is in the past */
-} CmdTimeExpRes;
+/* Used by hpersistCommand() */
+typedef enum SetPersistRes {
+    HFE_PERSIST_NO_FIELD =     -2,   /* No such hash-field */
+    HFE_PERSIST_NO_TTL =       -1,   /* No TTL attached to the field */
+    HFE_PERSIST_OK =            1
+} SetPersistRes;
+
+/* Used by hashTypeSetExpire() */
+typedef enum SetExpireTimeRes {
+    HFE_SET_NO_FIELD =         -2,   /* No such hash-field */
+    HFE_SET_NO_CONDITION_MET =  0,   /* Specified NX | XX | GT | LT condition not met */
+    HFE_SET_OK =                1,   /* Expiration time set/updated as expected */
+    HFE_SET_DELETED =           2    /* Field deleted because the specified time is in the past */
+} SetExpireTimeRes;
+
+/* Used by httlGenericCommand() */
+typedef enum GetExpireTimeRes {
+    HFE_GET_NO_FIELD =          -2,   /* No such hash-field */
+    HFE_GET_NO_TTL =            -1,   /* No TTL attached to the field */
+} GetExpireTimeRes;
```

Comment:
Why don't you use a single enum here? Error codes should be identical anyway.

Reply:
I aimed to clearly distinguish and specify, in one place, the expected subset of returned values for each command.